pkg/index/index.go

```diff
@@ -52,7 +52,7 @@ func (n *node) addChild(child *node) {
 type Index struct {
 	root        *node
 	lookupTable map[string]*node
-	m           sync.Mutex
+	m           sync.RWMutex
```

Comment:
Just saw this.  Interesting.  For readability.  I like it!   I'll fix this in my next PR.